db/migrate/20231114223004_add_my_settings_view_to_roles.rb

```diff
@@ -0,0 +1,48 @@
+class AddMySettingsViewToRoles < ActiveRecord::Migration[6.1]
+  class MiqUserRole < ActiveRecord::Base
+    has_and_belongs_to_many :miq_product_features, :join_table => :miq_roles_features, :class_name => "AddMySettingsViewToRoles::MiqProductFeature"
+  end
+
+  class MiqProductFeature < ActiveRecord::Base; end
+
+  def up
+    return unless MiqUserRole.exists?(:read_only => false)
+
+    say_with_time("Adding my_settings_view to custom user roles") do
+      my_settings_features = MiqProductFeature.where("identifier LIKE ?", "my_settings_%").where.not(:identifier => "my_settings_view")
+      my_settings_view = nil
+
+      affected_user_roles(my_settings_features).each do |user_role|
+        product_features = user_role.miq_product_features.map(&:identifier)
+
+        next if product_features.include?("my_settings") || product_features.include?("my_settings_view")
+
+        # This migration will likely run before the product feature is seeded so we need to manually insert it otherwise
+        my_settings_view ||= MiqProductFeature.create_with(
+          :identifier   => "my_settings_view",
```

Comment:
Can't use create because of this test:

`AddMySettingsViewToRoles#up when my_settings_view already exists updates custom user roles with a my_settings_* feature`

We need it find the existing one if the role already exists. This is for cases where developers may already have the role because core was merged before the migration was ready.  Edge case yea, but easy enough to handle.